hive-metastore/src/main/java/org/apache/iceberg/hive/HiveSchemaConverter.java

```diff
@@ -73,9 +75,13 @@ Type convertType(TypeInfo typeInfo) {
             return Types.BooleanType.get();
           case BYTE:
           case SHORT:
-            throw new IllegalArgumentException("Unsupported Hive type (" +
-                ((PrimitiveTypeInfo) typeInfo).getPrimitiveCategory() +
-                ") for Iceberg tables. Consider using INT/INTEGER type instead.");
+            if (autoConvert) {
+              return Types.IntegerType.get();
```

Comment:
can we add some logging so we can track where autoconversion happened? should be useful for debugging

Reply:
Good idea.
Added the comment